### DISCUSSION TITLE
AttributeError: 'Trainer' object has no attribute 'running_sanity_check"

### DISCUSSION THREAD

**Discussion by annalauralerede** [Lightning-AI/pytorch-lightning#11926] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
Hi,

I am trying to optimise the hyperparameters of my network using raytune. My implementation is pretty much based on this: 
https://docs.ray.io/en/master/tune/tutorials/tune-pytorch-lightning.html#selecting-a-scheduler

When I train my network using pre-set hyperparameters, it works smoothly. The problems come from the callback, so when I add the following line:
`TuneReportCallback({"loss":"val_loss"}, on="validation_end")`

I get the following error:
![Screenshot 2022-02-15 at 17 34 23](https://user-images.githubusercontent.com/57765003/154106370-c343c64c-b255-4bb7-86a3-9fdf2293492a.png)

Anyone knows how to solve this??
I don't think the problem is with my code as I haven't done anything different compared to the tutorial!

My code can be found here:
https://github.com/annalauralerede/anomaly-detection/blob/main/lstmae_pl_opt.py

**Comment by pvmilk** [comment, marked as answer]
I think it dues to this : https://github.com/ray-project/ray/issues/20741

As of ray[tune]==1.10.0, either downgrade your pytorch-lightning to 1.4.
Or upgrade your raytune to be compatible with pytorch-lightning 1.5+ ( the fix has been merged in this commit https://github.com/ray-project/ray/pull/20562).
`$ sudo pip install ray==1.11.0rc0`
